mlrun/runtimes/function.py

```diff
@@ -483,6 +485,52 @@ def _wait_for_function_deployment(self, db, verbose=False):
             logger.error("Nuclio function failed to deploy", function_state=state)
             raise RunError(f"function {self.metadata.name} deployment failed")
 
+    def with_node_selection(
+        self,
+        node_name: typing.Optional[str] = None,
+        node_selector: typing.Optional[typing.Dict[str, str]] = None,
+        affinity: typing.Optional[client.V1Affinity] = None,
+    ):
+        if not mlrun.mlconf.nuclio_version:
+            raise mlrun.errors.MLRunMissingDependencyError(
+                "Missing nuclio version for node selection "
+                "compatibility verification"
+            )
+        try:
+            # TODO: handle labels like stable, unstable, latest
+            parsed_nuclio_version = semver.VersionInfo.parse(
+                mlrun.mlconf.nuclio_version
+            )
+        except ValueError as exc:
+            logger.error(
+                "Unable to parse nuclio version",
+                nuclio_version=mlrun.mlconf.nuclio_version,
+            )
+            raise exc
```

Comment:
Same here, `nuclio_version` might be `stable` which is not semver, log warning and return